alpha/lib/model/LiveEntry.php

```diff
@@ -480,20 +480,37 @@ public function isCurrentlyLive($currentDcOnly = false)
 		if(!count($liveEntryServerNodes))
 			return false;
 
+		$primaryEsn = null;
+		$secondaryEsn = null;
+
 		/* @var LiveEntryServerNode $liveEntryServerNode*/
 		foreach($liveEntryServerNodes as $liveEntryServerNode)
 		{
-			/* @var WowzaMediaServerNode $serverNode*/
-			$serverNode = ServerNodePeer::retrieveActiveMediaServerNode(null, $liveEntryServerNode->getServerNodeId());
-			if($serverNode->getDc() == kDataCenterMgr::getCurrentDcId())
+			if ($liveEntryServerNode->getServerType() == EntryServerNodeType::LIVE_PRIMARY)
+			{
+				$primaryEsn = $liveEntryServerNode;
+			}
+			else if ($liveEntryServerNode->getServerType() == EntryServerNodeType::LIVE_BACKUP)
+			{
+				$secondaryEsn = $liveEntryServerNode;
+			}
+		}
+
+		if ($primaryEsn || $secondaryEsn)
+		{
+			if ($this->getExplicitLive() && !$this->canViewExplicitLive())
+				return false;
+			if ($primaryEsn)
+			{
+				if ($primaryEsn->getIsPlayableUser())
+					return true;
+			}
+			else if ($secondaryEsn && $secondaryEsn->getIsPlayableUser())
 			{
-				if ($this->getExplicitLive() && !$this->canViewExplicitLive() && !$liveEntryServerNode->getIsPlayableUser())
-					return false;
 				return true;
 			}
 		}
-		
-		return !$currentDcOnly;
+		return false;
```

Comment:
This entire block can be replace with shorter syntax - 
return $primaryEsn && $primaryEsn->getIsPlayableUser() || 
           $secondaryEsn && $secondaryEsn->getIsPlayableUser();